src/runtime/builtin_modules/builtins.cpp

```diff
@@ -1478,6 +1478,7 @@ void setupBuiltins() {
     enumerate_cls->giveAttr("__hasnext__",
                             new BoxedFunction(boxRTFunction((void*)BoxedEnumerate::hasnext, BOXED_BOOL, 1)));
     enumerate_cls->freeze();
+    enumerate_cls->tp_iter = BoxedEnumerate::iter;
```

Comment:
I think sharing the code between the capi (`tp_iter`) and c++ (`__iter__` attribute) versions is fine since none of these functions should throw any exceptions, but maybe we could add noexcept to these functions?